middlewares/redirect_as2.ts

```diff
@@ -0,0 +1,68 @@
+import type { Middleware } from "../core/server.ts";
+
+export type Options = (url: URL) => Promise<URL> | URL | undefined | void;
+
+export interface CommonOptions {
+  /**
+   * User domain.
+   * @default
+   * ```ts
+   * new URL(req.url).host
+   * ```
+   * @example
+   * ```ts
+   * site.options.location.host
+   * ```
+   */
+  host?: URL["host"];
+}
+
+export interface HatsuOptions extends CommonOptions {
+  /**
+   * Hatsu instance.
+   * @default `hatsu.local`
+   */
+  instance: URL["host"];
+}
+
+export interface BridgyFedOptions extends CommonOptions {
+  /**
+   * BridgyFed instance.
+   * @default `fed.brid.gy`
+   */
+  instance?: URL["host"];
+}
+
+export const hatsu = (options: HatsuOptions): Options => (url: URL) => {
```

Comment:
Good idea to provide some functions to rewrite the URL.
In this case, I think we don't need an options object but only the instance URL.

I'm thinking of something like this:

```js
server.use(redirect_as2({
  redirect: hatsu("https://hatsu.local"),
}));
```

Reply:
> In this case, I think we don't need an options object but only the instance URL.

The `rewriteUrl` function also uses the `host` param from `CommonOptions`.

Reply:
Ah, I see. And what about this?

```js
server.use(redirect_as2({
  location: new URL("https://example.com"), // If it's not specified, use request.url
  redirect: hatsu("https://hatsu.local"),
}));
```

And the redirect functions should do something like this:

```js
export function hatsu(instanceURL) {
  return (requestURL, instanceURL, siteURL) {
    // code here
  }
}
```

My point is to move the common options to the middleware. `hatsu` (or `bridgy`) is only a way to generate the redirection url, it should have only the hatsu specific configuration.

Reply:
> My point is to move the common options to the middleware. `hatsu` (or `bridgy`) is only a way to generate the redirection url, it should have only the hatsu specific configuration.

Okay, I've moved `host` option to `Options`.